In [ ]:
 
#1. Predict the output for the following code snippets:
print(0.1+0.2)
print("1.8"+"2")
print(87>78)
print((0.1+0.2)==0.3)
print("predict " "output" ,".....")

0.30000000000000004
1.82
True
False
predict output .....


In [ ]:
 
#2. Predict the output for the following code snippets
print("Prints","multiple ","messages\n")
print("Concatenate "+" two strings\n")
print(5+6+" adds two numbers\n")

Prints multiple  messages

Concatenate  two strings



TypeError: ignored

In [ ]:
 
#3. Correct the given code snippets (only one change is allowed)
print("ba"+"na"*2)
print("c:\\naresh\\raju\\abhi")

banana
c:\naresh\raju\abhi


In [ ]:
 
#4. Correct the given code snippets
'Welcome to python traing program'[-4::-4]
#print('A series of characters designated as one object knows as a string'[2:18:5])

'g anytoW'

In [ ]:
 
#4.b
 print('A series of characters designated as one object knows as a string'[2:18:5])

sscc


In [ ]:
 
#5. Predict the output for the given code snippet and justify the result.
print('A series of characters designated as one object known as a string'[::-1][4::3])
print("-----------")
print("Welcome to Python traing program"[3:10][::-1])

taawkcbe  tgestrcoeeA
-----------
ot emoc


 From above question 
1.in first print statement we can say that when[::-1]this is mode where it prints in reverse order and next[4::3]this is the case after printing reverse order then again it starts from 4th uppercase to 3 steps forward
2.in second print statement it prints as it is 
3.in third print statement it prints from 3 element to 10 element after that the same it prints in reverse order

In [ ]:
 
#6.a. Predict the output for the given code snippet and justify the result
Str1 = True
x = 5 > 3
print (Str1 == x)
y = 5 > 8
print (Str1 == y)

True
False


 In first case it is true beacuse 5 is greater than 3
In second case it is false beacuse 5 is less than 8

In [ ]:
 
#6.b
num = 7
Name = "Michael Jackson"
Sear_num = Name.find('el')
print(num > Sear_num)

True


In [ ]:
 
#7. Write a python code to get desired output
#Input string : p_phrase = "was it a car or a cat I saw"
 #Output string : WAS I TAC A RO RAC A TI SAW
P_pharse = 'was it a car or a cat i saw'
P_pharse.upper()[::-1]

'WAS I TAC A RO RAC A TI SAW'

In [ ]:
 
#8. Write a python code to print integer value : 946
#Input string : A='1934567'
A = '1934567'
print (A[1:6:2])

946


In [ ]:
 
#8. Write a python code to print integer value : 946
Input string : A='1934567'
fake_pharse='Fake news has a knack for spreading like'
fake_pharse=fake_pharse.upper()[15:40][::-1]
print(fake_pharse.split())

['EKIL', 'GNIDAERPS', 'ROF', 'KCANK']


In [ ]:
 
#10.Write a python code to get desired output
#msg1 = " Facebook already uses AI to Filter Fake stories from the feeds of users "
#out = [Facebook , already, uses, AI, to, Filter, Fake, stories, from, the, feeds, of, users ]
msg1 = "Facebook already uses Al to filter Fake stories from the feeds of users"
msg1.split()

['Facebook',
 'already',
 'uses',
 'Al',
 'to',
 'filter',
 'Fake',
 'stories',
 'from',
 'the',
 'feeds',
 'of',
 'users']

In [ ]:
#11.Predict the output for given code snippet .Justify the result .
msg2 = 'Welcome to sr engineering college'
X = msg2.count('o')
Y = msg2.count('r')
print(X)
print(Y)
msg2[Y**X:(X**Y+X+Y):][::-1]

3
2


' rs ot'

 It has counted the occurrence of o in msg2 and occurrence of r in msg2
Then[8:(14):][::-1] this is implemented

In [ ]:
 
#12.Predict the output for given code snippet .Justify the result
num1,num2 = "94","30"
data = "As per census 2011,Gender ratio of India is 943 females per 1000 males"
num1+num2[0]in data
 
#b
print(data[:45],print(int(num1)+int(num2)))

124
As per census 2011,Gender ratio of India is 9 None


It prints 124 bcz (94+30)
and in print statement it prints only it prints up to the 44th elements

In [ ]:
 
#13.Develop a program that calculates the energy needed to heat water from 
#an initial temperature to a final temperature. Your program should prompt 
#the user to enter the amount of water in kilograms and the initial and final 
#temperatures of the water. The formula to compute the energy is 
#Q = M * (finalTemperature – initialTemperature) * 4184.
#Where M is the weight of water in kilograms, temperatures are in degrees 
#Celsius, and energy Q is measured in joules. 
M=float(input('enter the amount of water'))
intial_temperature=float(input('enter the intial temperature'))
final_temperature=float(input('enter final temperature'))
Q=M*(final_temperature-intial_temperature)*4184
print(Q)

enter the amount of water56
enter the intial temperature78
enter final temperature90
2811648.0


In [ ]:
 
#14.You are required to generate a word similar to ZOO, X and Y that 
#denote the number of Zs and Os respectively. The generated word similar 
#to ZOO where Y= 2 * X.
x=int(input('N : '))
Z='Z'*x
O='O'*x*2
print(Z+O)

N : 5
ZZZZZOOOOOOOOOO


In [ ]:
 
#15.Read two integers X ,Y from STDIN and print three lines where:
#1. The first line print ,X to the power of Y (store the results into 
#pow)
#2. The second line prints the floor division between pow and 
#product of X and Y (store the results into div).
#3. The third line prints the exclusive or (^) between div and sum 
#of X and Y
x=int(input("enter the value of x : "))
y=int(input("enter the value of y : "))
pow=x**y
div=x//y
print(pow)
print(div)
print(div^(x+y))

enter the value of x : 4
enter the value of y : 6
4096
0
10
